### PR TITLE
Add SPIR-V friendly translation for OpLoad and OpStore

### PR DESCRIPTION
**Pull request by MrSidims**
This patch should unlock some possibilities to manually write SPIR-V code. 

Signed-off-by: Dmitry Sidorov <dmitry.sidorov@intel.com>

### CODE REVIEW

File: lib/SPIRV/SPIRVWriter.cpp [hunk 1]
```diff
@@ -4379,6 +4379,25 @@ LLVMToSPIRVBase::transBuiltinToInstWithoutDecoration(Op OC, CallInst *CI,
...
+    assert(CI->arg_size() > 0 && "Expected at least 1 operand for OpLoad call");
+    for (size_t I = 1; I < CI->arg_size(); ++I)
+      MemoryAccess.push_back(
+          cast<ConstantInt>(CI->getArgOperand(I))->getZExtValue());
+    return BM->addLoadInst(transValue(CI->getArgOperand(0), BB), MemoryAccess,
+                           BB);
+  }
+  case OpStore: {
+    std::vector<SPIRVWord> MemoryAccess;
+    assert(CI->arg_size() > 1 && "Expected at least 2 operand for OpLoad call");
```
Comment (Pennycook) [inline lib/SPIRV/SPIRVWriter.cpp:4393]:
```suggestion
    assert(CI->arg_size() > 1 && "Expected at least 2 operand for OpStore call");
```

Comment (MrSidims) [reply lib/SPIRV/SPIRVWriter.cpp]: Thanks!

**Comment by MrSidims** [comment]
@AlexeySotkin @AlexeySachkov @svenvh could you please take a look?

File: test/spirv-load-store.ll [hunk 2]
```diff
@@ -0,0 +1,68 @@
+; Translate SPIR-V friendly OpLoad and OpStore calls
+; RUN: llvm-as %s -o %t.bc
+; RUN: llvm-spirv %t.bc -spirv-text -o %t
+; RUN: FileCheck < %t %s
+; RUN: llvm-spirv %t.bc -o %t.spv
```
Comment (svenvh) [inline test/spirv-load-store.ll:5]: Maybe add a `spirv-val` step too?

Comment (MrSidims) [reply test/spirv-load-store.ll:5]: I've added it in the original commit, but existence of __spirv_BuiltInGlobalInvocationId results in a failure (if I remember correctly in a known SPIR-V 1.4 entry validation failure). I'll rewrite the test, removing __spirv_BuiltInGlobalInvocationId and adding spirv-val.

**Review by svenvh — APPROVED** [review]

**Pull request merged**